### DISCUSSION TITLE
Microfrontend setup error: ScriptExternalLoadError: Loading script failed.

### DISCUSSION THREAD

**Discussion by dagaranupam** [webpack/webpack#19085] [Q&A]
Hi,

I have a microfrontend app. I was enabling optimisations in webpack for my apps to reduce the bundle size. I did so and was successfully able to reduce the bundle size for both of my parent and child app. But now when I run my parent app along with the child app, I am getting the error `Initialization of sharing external failed: ScriptExternalLoadError: Loading script failed.
(missing: https://localhost:3008/remoteEntry.js)`.
When I check the network tab, I can see that the remoteEntry.js call has succeeded with ok status. I can confirm that the url set in the `remotes` of webpack configuration is correct. `If I remove the `splitChunks` property from optimisation of my child app, then it works fine.

I want to highlight that my child app `exposes` 2 files `./App`, `./routes`.  The webpack config of parent app has the following rule
```
{
        test: /routes\.js$/,
        use: [
          {
            loader: path.resolve("routeloader.js"),
            options: {},
          },
        ],
      }
```
The `routeloader.js` imports the child app routes and injects in the parent app routes.

I don't want to remove the chunking from child apps as it is increasing the bundle size. Looking forward to know what configuration I have been doing wrong.
<img width="965" alt="Screenshot 2024-12-19 at 14 09 12" src="https://github.com/user-attachments/assets/1421bccb-d0b2-4151-9bd4-308c07c663fd" />

**Comment by alexander-akait** [comment]
Can you show me `splitChunks` configuration?

**Reply by dagaranupam** [reply]
Parent App `splitChunks`:
```
  optimization: {
    minimize: true,
    minimizer: [new TerserPlugin()],
    usedExports: true,
    splitChunks: {
      cacheGroups: {
        vendors: {
          test: /[\\/]node_modules[\\/]/,
          chunks: "all"
        },
      },
      chunks: 'all',
    },
  },
```

Child App `splitChunks`:
```
    optimization: {
        minimize: true,
        minimizer: [new TerserPlugin()],
        usedExports: true,
        splitChunks: {
            cacheGroups: {
                vendors: {
                    test: /[\\/]node_modules[\\/]/,
                    name: "vendor",
                    chunks: "all"
                },
            },
            chunks: 'all',
        },
    },
```

**Reply by alexander-akait** [reply]
You can not use 'all' right now - https://github.com/webpack/webpack/issues/18809, try to specify specific chunks

**Reply by dagaranupam** [reply]
Could you provide me with sample way of specifying chunks in this scenario? I did try some other ways of specifying cache groups but the chunks was set to `all` in that scenario. Should it be set to async, initial or there must be different way of specifying cacheGroups?

**Reply by alexander-akait** [reply]
No, use it as a function - https://webpack.js.org/plugins/split-chunks-plugin/#splitchunkschunks and experiment with this, unfortunately without a real structure I can't give advice on the right setting

**Reply by alexander-akait** [reply]
The main idea is that the module should not go into the initial chunks of the main application
